pkg/api/jokes_test.go

```diff
@@ -0,0 +1,222 @@
+package api
+
+import (
+	"bufio"
+	"bytes"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+
+	file_storage "github.com/DanilLagunov/jokes-api/pkg/storage/file-storage"
+	"github.com/DanilLagunov/jokes-api/pkg/views"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetJokes(t *testing.T) {
+	storage := file_storage.NewFileStorage("../../test-data/test_jokes.json")
+	template := views.NewTemptale("../../templates/")
+	h := NewHandler(storage, template)
+
+	recorder := httptest.NewRecorder()
+	req := httptest.NewRequest(http.MethodGet, "/jokes", nil)
+
+	h.getJokes(recorder, req)
+	data, err := os.Open("../../test-data/index_test.html")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer data.Close()
+
+	reader := bufio.NewReader(data)
```

Comment:
you overcomplicate a simple task. It's easy to use `os.ReadFile` to get content of the file and to use `recorder.Body.Bytes()` to get response.